esphome/components/debug/debug_esp32.cpp

```diff
@@ -85,7 +85,61 @@ std::string DebugComponent::get_reset_reason_() {
     defined(USE_ESP32_VARIANT_ESP32S3) || defined(USE_ESP32_VARIANT_ESP32C6)
     case RTC_SW_CPU_RESET:
 #endif
-      reset_reason = "Software Reset CPU";
+      switch (esp_reset_reason()) {
+        case ESP_RST_UNKNOWN:
+          reset_reason = "Reset reason can not be determined";
+          break;
+        case ESP_RST_POWERON:
+          reset_reason = "Reset due to power-on event";
+          break;
+        case ESP_RST_EXT:
+          reset_reason = "Reset by external pin";
+          break;
+        case ESP_RST_SW:
+          reset_reason = "Software reset via esp_restart";
+          break;
+        case ESP_RST_PANIC:
+          reset_reason = "Software reset due to exception/panic";
+          break;
+        case ESP_RST_INT_WDT:
+          reset_reason = "Reset (software or hardware) due to interrupt watchdog";
+          break;
+        case ESP_RST_TASK_WDT:
+          reset_reason = "Reset due to task watchdog";
+          break;
+        case ESP_RST_WDT:
+          reset_reason = "Reset due to other watchdogs";
+          break;
+        case ESP_RST_DEEPSLEEP:
+          reset_reason = "Reset after exiting deep sleep mode";
+          break;
+        case ESP_RST_BROWNOUT:
+          reset_reason = "Brownout reset (software or hardware)";
+          break;
+        case ESP_RST_SDIO:
+          reset_reason = "Reset over SDIO";
+          break;
+#if defined(USE_ESP_IDF) && ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 1, 0)
```

Comment:
```suggestion
#ifdef USE_ESP32
#if (ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(4, 4, 5))
```
- The preprocessor checks should be divided into multiple lines to work properly.
- Is there any specific need setting minimum version to 5.1? I see the used constants are available since version [4.2](https://github.com/espressif/esp-idf/blob/release/v4.2/components/esp_system/include/esp_system.h). However, in my tests it's actually compiles and works since version 4.4.5 that contains all necessary bug fixes. So I suggest putting 4.4.5 here.

Reply:
Not all the constants are available prior to v5.1. The ones above this #if are the ones available since earlier versions.

Reply:
And to be honest, I think this should be reversed in a way this `esp_reset_reason` first and only when that is unknown to use the other method.
Any thoughts?

Reply:
I've reverted this to v5.1.0 as those advanced options are not available before IDF v5.1.0 and therefore causes error when compiling with ESP-IDF default version (4.4.8).